hooks/useDelayResetState.ts

```
@@ -0,0 +1,36 @@
+ */
+export function useDelayResetState<T>(
+  initialState: T,
+  delay: number
+): [T, Dispatch<SetStateAction<T>>, Dispatch<SetStateAction<T>>] {
+  const [resetState, setResetState] = useState(initialState);
+  const [state, setState] = useState(initialState);
+
+  const setDelayResetState = useCallback(
+    (value: SetStateAction<T>) => {
+      if (typeof value === 'function') {
+        value = (value as Function)(state);
```

Biome (1.9.4)

[error] Don't use 'Function' as a type.

Prefer explicitly define the function shape. This type accepts any function-like value, which can be a common source of bugs.

(lint/complexity/noBannedTypes)